addons/account/models/account_move.py

```diff
@@ -3209,8 +3209,6 @@ def add_file_data_results(file_data, invoice):
                             invoices |= invoice
                             current_invoice = self.env['account.move']
                             add_file_data_results(file_data, invoice)
-                        if extend_with_existing_lines:
-                            return attachments_by_invoice
```

Comment:
I discussed this with LAS - we should maybe change this only when `new=True`
So we'd suggest amending L3196 to
`extend_with_existing_lines = file_data.get('process_if_existing_lines') and not new`

Reply:
(actually, I'm having second thoughts about this! - since we also don't want attachments to get dropped when the customer sends several attachments to an existing invoice)